Lib/test/test_frame.py

```diff
@@ -328,6 +328,7 @@ def f():
             next(g)
             # Move all objects to the oldest generation, and tell the GC to run
             # on the *very next* allocation:
+            sys._clear_internal_caches()
             gc.collect()
             gc.set_threshold(1, 0, 0)
```

Comment:
Nit... this probably makes more sense after the `gc` stuff:

```suggestion
            gc.collect()
            gc.set_threshold(1, 0, 0)
            sys._clear_internal_caches()
```